transitions/generate_patient_id_on_patients.js

```diff
@@ -0,0 +1,11 @@
+var transitionUtils = require('./utils');
+
+module.exports = {
+  filter: function(doc) {
+    return doc.type === 'person' &&
```

Comment:
Also added a functional test for id generation as a minor sanity check because there was no testing around it afaict: https://github.com/medic/medic-sentinel/pull/115  and confirmed independently that a id_format of '1111' does indeed give us a pool of 10000 ids.